rest/handler.go

```diff
@@ -584,17 +584,29 @@ func (h *handler) writeJSONStatus(status int, value interface{}) {
 		json.Indent(&buffer, jsonOut, "", "  ")
 		jsonOut = append(buffer.Bytes(), '\n')
 	}
+	h.writeRawJSONStatus(status, jsonOut)
+}
+
+// writeRawJSONStatus writes the given byte slice as a JSON response.
+// If status is nonzero, the header will be written with that status.
+func (h *handler) writeRawJSONStatus(status int, b []byte) {
+	if !h.requestAccepts("application/json") {
```

Comment:
Is this check cheap enough that we can do it twice when writeJSONStatus is called?  Or can it be moved into writeRawJSON prior to calling here?

Reply:
It's fairly cheap (a map lookup and a bit of string comparisons). My main motivation for keeping it around above was to avoid the marshalling in `writeJSONStatus`, but I've rearranged the code to call into a common method between both.